datajunction-clients/python/datajunction/nodes.py

```diff
@@ -276,31 +327,62 @@ def set_column_attributes(
         return self.dj_client._set_column_attributes(self.name, column_name, attributes)
 
 
+@dataclass
 class Source(Node):
     """
     DJ source node
     """
 
     type: str = "source"
-    catalog: str
-    schema_: str
-    table: str
-    columns: Optional[List[models.Column]]
+    catalog: Optional[str] = None
+    schema_: Optional[str] = None
+    table: Optional[str] = None
+
+    def to_dict(self, exclude: Optional[List[str]] = None) -> Dict[str, Any]:
```

Comment:
I think calls to this function could be replaced by `dataclasses.asdict`, no?

Reply:
Yes...

Reply:
Actually no. Dataclasses fail miserably on `asdict()` when they have any nested dataclasses. That's why I needed this to_dict().